src/Errors/EleventyErrorHandler.js

```diff
@@ -91,9 +90,9 @@ class EleventyErrorHandler {
 		while (ref) {
 			let nextRef = ref.originalError;
 
-			// Nunjucks wraps errors and puts the original in error.cause
-			if (nextRef?.cause?.originalError) {
-				nextRef = nextRef.cause.originalError;
+			// Unwrap cause from error and assign it to what Eleventy expects
```

Comment:
Have you looked at

https://github.com/11ty/eleventy/commit/9f957abb624270bc33fcc1579436c7f771c83c86#diff-bf3ab5639a6c77684327c6f227d9ea440e361ebfb365ebfe93effde8c80c4926

to make sure it behaves well with Nunjucks?

Or, what template engines have you tested this with?

Reply:
> Have you looked at https://github.com/11ty/eleventy/commit/9f957abb624270bc33fcc1579436c7f771c83c86#diff-bf3ab5639a6c77684327c6f227d9ea440e361ebfb365ebfe93effde8c80c4926 to make sure it behaves well with Nunjucks?

See my original comment—the example I'm using **is** Nunjucks. This change improves the error; previously the error specifying `Error with Nunjucks shortcode...` wasn't getting logged.

I also just tested with JavaScript and Liquid templates—this change doesn't degrade logging of their errors at all (although the line referencing this is a "shortcode error" doesn't get logged for them either). This edit only improves Nunjucks and Custom templates that chain errors via `cause`.

FFIW, as of Node 18 (Eleventy's minimum supported version), [the convention is to rethrow errors with `cause` passed as part of the `ErrorOptions` object](https://nodejs.org/docs/latest-v18.x/api/errors.html#new-errormessage-options). That said, should Eleventy's error handler be refactored to align with that standard instead of the non-standard `.orignalError` property currently used?

Reply:
> That said, should Eleventy's error handler be refactored to align with that standard instead of the non-standard .orignalError property currently used?

That's for @zachleat to decide. (Perhaps worth a new issue?)

Reply:
Absolutely, filed that here: #3582